### PR TITLE
Fix panic when scheduler is not initialized and job API is used.

### PR DESCRIPTION
**Pull request by artursouza**
# Description

Fixes https://github.com/dapr/dapr/issues/7897

New integration test reproduces the bug gracefully. In this case, the check for `c == nil` was required. It means that the `.Next()` function is called even when the instance is nil.

## Issue reference

Please reference the issue this PR will close: https://github.com/dapr/dapr/issues/7897

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/runtime/scheduler/clients/clients.go [hunk 1]
```diff
@@ -62,6 +62,12 @@ func New(ctx context.Context, opts Options) (*Clients, error) {
 
 // Next returns the next client in a round-robin manner.
 func (c *Clients) Next() schedulerv1pb.SchedulerClient {
+	// This is called even when no scheduler is initialized.
+	// Keeping these nil checks here as a central location instead of spreading "ifs".
+	if c == nil || c.clients == nil || len(c.clients) == 0 {
+		return &schedulerClientNotInitializedInstance
```
Comment (JoshVanL) [inline pkg/runtime/scheduler/clients/clients.go:68]: Does it make more sense to not return an error here instead? Or better, also pass a context whereby it waits for client to become available. We return the context error when it has been cancelled.

File: pkg/runtime/scheduler/clients/schedulerclientnotinit.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
+/*
```
Comment (JoshVanL) [inline pkg/runtime/scheduler/clients/schedulerclientnotinit.go:1]: Rename file to `noinit` as prefix is superfluous through package import name.

**Review by JoshVanL — CHANGES_REQUESTED** [review]

File: pkg/runtime/scheduler/clients/clients.go (hunk 1 above)
Comment (artursouza) [reply pkg/runtime/scheduler/clients/clients.go:68]: I am keeping this change minimal for the sake of this release.

File: pkg/runtime/scheduler/clients/schedulerclientnotinit.go (hunk 2 above)
Comment (artursouza) [reply pkg/runtime/scheduler/clients/schedulerclientnotinit.go:1]: Sure.

**Review by yaron2 — APPROVED** [review]

File: pkg/runtime/scheduler/clients/clients.go (hunk 1 above)
Comment (artursouza) [reply pkg/runtime/scheduler/clients/clients.go:68]: If you want to make another PR in master improving this, I am all in. For 1.14, I recommend us being minimalistic.

**Pull request merged**